### ISSUE TITLE
Plugin.initializeGoApplicationAccessor called multiple times

### ISSUE THREAD

**Issue by danielsomerfield** [gocd/gocd#2035]
##### Issue Type

- Bug Report
##### Summary

Wrote a plugin and found that I was seeing some strange behavior because initializeGoApplicationAccessor() was being called three times on the same instance of the plugin, with the same GoApplicationAccessor. Is this expected behavior?
##### Environment

/api/go/support is returning a 404 for some reason, but I can tell you this:

Running the docker instance in development mode from here: https://github.com/gocd/gocd-docker
###### Basic environment details

Go Version: Go Version: 16.1.0(2855-ada9b36174cc069c860e7fa032cbf2857f135cfb)

JAVA Version:
java version "1.7.0_95"
OpenJDK Runtime Environment (IcedTea 2.6.4) (7u95-2.6.4-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)

OS:

Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:    14.04
Codename:   trusty
##### Steps to Reproduce

Create plugin with log message in initializeGoApplicationAccessor().  
##### Expected Results

See the log message once.
##### Actual Results

Saw it three times, with the same plugin and accessor instance.
##### Any other info

You can see the plugin I'm working on at the link below, but really, any plugin will probably do:
https://github.com/danielsomerfield/go-strong-auth-plugin

**Comment by arvindsv** [comment]
@danielsomerfield: I checked with a fresh server, with your plugin itself, and all I see in the log file is this:

```
2016-03-09 11:23:25,960  INFO [Thread-69] StrongAuthPlugin:52 - Creating plugin: com.thoughtworks.go.strongauth.StrongAuthPlugin@175704df
2016-03-09 11:23:25,967  INFO [Thread-69] StrongAuthPlugin:52 - Initializing accessor com.thoughtworks.go.strongauth.StrongAuthPlugin@175704df with com.thoughtworks.go.plugin.infra.DefaultGoApplicationAccessor@2f5c2bae.
2016-03-09 11:23:26,349  INFO [pool-2-thread-1] ChangeMonitor:52 - Value changed from Optional.absent() to Optional.of(PluginConfiguration{principalSourceFilePath='/etc/go/passwd'})
2016-03-09 11:23:26,351  INFO [pool-2-thread-1] ChangeMonitor:52 - Instance: com.thoughtworks.go.strongauth.util.ChangeMonitor$2@76fff18d
```

I don't see it happening three times. I looked at the code and it calls it once for every `GoPlugin` instance there is.

**Comment by danielsomerfield** [comment]
Hmmm. That's odd. I'll try and repo again. Maybe there is something on my
side.

I did add a work-around that fixed it to check if the GoApplicationAccessor was
already set and skip if it's not null.

**Comment by arvindsv** [comment]
> I did add a work-around that fixed it to check if the GoApplicationAccessor was
already set and skip if it's not null.

Oh. Maybe that is what is hiding the three calls. I'll remove that from your code and see what happens.

**Comment by ketan** [comment]
@arvindsv — we noticed this unintended behaviour while working on the docker elastic agent plugin and put in a fix for it, but it's not on master yet.

Instead of lazily initializing the plugin we seem to invoke `initializeGoApplicationAccessor` before every message is sent to it.

<img width="613" alt="screen shot 2016-03-09 at 10 37 25 pm" src="https://cloud.githubusercontent.com/assets/10598/13643598/ab6fc01e-e647-11e5-94be-4e6831131a93.png">

**Comment by arvindsv** [comment]
@ketan: I see. You're right. I didn't see it is in `submitTo`. That seems wrong. Ok, that can explain the issue. Then, I won't spend more time on it. I'll wait for your fix to hit master.

**Comment by ketan** [comment]
This was resolved via e2ffb0dfd925db38964e2213659baca2d9e1b76f in 16.8.0
